packages/e2e-tests/specs/experiments/blocks/comments-query.test.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	activateTheme,
+	createNewPost,
+	insertBlock,
+	pressKeyTimes,
+	publishPost,
+	setOption,
+	trashAllComments,
+} from '@wordpress/e2e-test-utils';
+
+describe( 'Comment Query Loop', () => {
+	beforeAll( async () => {
+		await activateTheme( 'emptytheme' );
+		await setOption( 'page_comments', true, 'options-discussion.php' );
+		await setOption( 'comments_per_page', '1', 'options-discussion.php' );
+		await createNewPost();
+	} );
+	it( 'Pagination links are working as expected', async () => {
+		// Insert the Query Comment Loop block.
+		await insertBlock( 'Comments Query Loop' );
+		// Insert the Comment Loop form.
+		await insertBlock( 'Post Comments Form' );
+		await publishPost();
+		// Visit the post that was just published.
+		await page.click(
+			'.post-publish-panel__postpublish-buttons .is-primary'
+		);
+
+		// Create three comments for that post.
+		for ( let i = 0; i < 3; i++ ) {
+			await page.waitForSelector( 'textarea#comment' );
+			await page.click( 'textarea#comment' );
+			await page.type(
+				`textarea#comment`,
+				`This is an automated comment - ${ i }`
+			);
+			await pressKeyTimes( 'Tab', 1 );
+			await page.keyboard.press( 'Enter' );
+			await page.waitForNavigation();
+		}
```

Comment:
Okay, let's keep this here for now. We can consider extracting it into a util once we find that we need it elsewhere 👍 
(Maybe add a comment to that effect? Like a `// TODO: ...`?)